lib/engine/game/base.rb

```diff
@@ -871,6 +872,17 @@ def route_trains(entity)
         entity.runnable_trains
       end
 
+      # Revenue bonuses that should be added to the list of train revenues
+      # displayed in the route selector. This method should return an array of
+      # hashes, with each item having these keys:
+      #  - id: The label to show in the `train` column of the route selector.
+      #  - used: Text to show in the `used` column.
+      #  - revenue: The bonus revenue.
+      #  - route: Text to show in the `route` column.
+      def revenue_bonuses(_entity)
```

Comment:
is this necessary?

Reply:
This is the model side of the change you queried in your [comment](https://github.com/tobymao/18xx/pull/8578/files/a441fe31534cee4e17fd4e556a1d3009435a6c66#r1045123208) on the changes in `assets/app/view/game/route_selector.rb`.

The default implementation is to return an empty array. Unless this method is overridden the game behaviour will not change.

Reply:
This has been removed.